code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/InputBoxNotificationBuilder.kt

```diff
@@ -0,0 +1,297 @@
+        val imageUri =
+            if (!pushTemplate.isFromIntent) pushTemplate.imageUrl else pushTemplate.feedbackImage
+        val pushImage = PushTemplateImageUtil.downloadImage(cacheService, imageUri)
+
+        if (pushImage != null) {
+            expandedLayout.setImageViewBitmap(R.id.expanded_template_image, pushImage)
+        } else {
+            Log.trace(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "No image found for input box push template."
+            )
+            expandedLayout.setViewVisibility(R.id.expanded_template_image, View.GONE)
+        }
+
+        val expandedBodyText =
+            if (pushTemplate.isFromIntent) pushTemplate.feedbackText else pushTemplate.expandedBodyText
+        val collapsedBodyText =
+            if (pushTemplate.isFromIntent) pushTemplate.feedbackText else pushTemplate.body
+        smallLayout.setTextViewText(R.id.notification_title, pushTemplate.title)
+        smallLayout.setTextViewText(R.id.notification_body, collapsedBodyText)
+        expandedLayout.setTextViewText(R.id.notification_title, pushTemplate.title)
+        expandedLayout.setTextViewText(
+            R.id.notification_body_expanded, expandedBodyText
+        )
+
+        // add an input box to capture user feedback if the push template is not from an intent
+        if (pushTemplate.isFromIntent || pushTemplate.inputBoxReceiverName.isNullOrEmpty()) {
+            return notificationBuilder
+        }
+
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Adding an input box to capture text input. The input box receiver name is ${pushTemplate.inputBoxReceiverName}."
+        )
+        addInputTextAction(
+            context,
+            broadcastReceiverClass,
+            notificationBuilder,
+            channelIdToUse,
+            pushTemplate
+        )
+
+        return notificationBuilder
+    }
+
+    /**
+     * Adds an input text action for the notification.
+     *
+     * @param context the application [Context]
+     * @param broadcastReceiverClass the [BroadcastReceiver] class to use as the broadcast receiver
+     * @param builder the [NotificationCompat.Builder] to attach the action buttons
+     * @param channelId the [String] containing the channel ID to use for the notification
+     * @param pushTemplate the [InputBoxPushTemplate] object containing the input box push template data
+     * button is pressed
+     */
+    private fun addInputTextAction(
+        context: Context,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?,
+        builder: NotificationCompat.Builder,
+        channelId: String,
+        pushTemplate: InputBoxPushTemplate
+    ) {
+        val inputHint =
+            if (!pushTemplate.inputTextHint.isNullOrEmpty()) pushTemplate.inputTextHint else DEFAULT_REPLY_LABEL
```

Comment:
recommend flipping the if condition for readability
`if(pushTemplate.inputTextHint.isNullOrEmpty()) .... else ...`